controllers/bsl_test.go

```diff
@@ -1666,9 +1697,191 @@ func TestDPAReconciler_ensureBackupLocationHasVeleroOrCloudStorage(t *testing.T)
 			r := &DPAReconciler{
 				Scheme: scheme,
 			}
-			if err := r.ensureBackupLocationHasVeleroOrCloudStorage(tt.dpa); (err != nil) != tt.wantErr {
-				t.Errorf("ensureBSLProviderMapping() error = %v, wantErr %v", err, tt.wantErr)
+			for _, bsl := range tt.dpa.Spec.BackupLocations {
+				if err := r.ensureBackupLocationHasVeleroOrCloudStorage(&bsl); (err != nil) != tt.wantErr {
+					t.Errorf("ensureBSLProviderMapping() error = %v, wantErr %v", err, tt.wantErr)
+				}
 			}
+
+		})
+	}
+}
+
+func TestDPAReconciler_ensurePrefixWhenBackupImages(t *testing.T) {
+	tests := []struct {
+		name        string
+		dpa         *oadpv1alpha1.DataProtectionApplication
+		wantErr     bool
+		expectedErr string
+	}{
+		{
+			name: "If DPA CR has CloudStorageLocation without Prefix defined with backupImages enabled, error case",
+			dpa: &oadpv1alpha1.DataProtectionApplication{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-DPA-CR",
+					Namespace: "test-ns",
+				},
+				Spec: oadpv1alpha1.DataProtectionApplicationSpec{
+					BackupLocations: []oadpv1alpha1.BackupLocation{
+						{
+							CloudStorage: &oadpv1alpha1.CloudStorageLocation{
+								CloudStorageRef: corev1.LocalObjectReference{
+									Name: "testing",
+								},
+								Prefix: "",
+							},
+						},
+					},
+					Configuration: &oadpv1alpha1.ApplicationConfig{
+						Velero: &oadpv1alpha1.VeleroConfig{
+							DefaultPlugins: []oadpv1alpha1.DefaultPlugin{},
+						},
+					},
+					BackupImages: pointer.Bool(true),
+				},
+			},
+			wantErr:     true,
+			expectedErr: "BackupLocation must have cloud storage prefix when backupImages is not set to false",
```

Comment:
Would it make sense to create type of error within controllers/*.go where the strings used in errors are stored? This way we don't have to compare against string here, but the string taken from that type? In case string message changes we don't have to update it in few places.

Reply:
ya this one makes sense to all the messages strings in all the test files, to be converted to Types.
We can have a separate PR for this one. I agree. Let me know if you want me to create one.

Reply:
Ok so let's not this be part of current PR, but I think it makes sense to keep all the error types / strings separate. Thinking here of something like:

```
type OadpSeverity string

const (
    Error       OadpSeverity = "error"
    Warning     OadpSeverity = "warning"
    Info        OadpSeverity = "info"
    Debug       OadpSeverity = "debug"
    Deprecation OadpSeverity = "deprecation"
)

type CustomMessage struct {
    OadpMessageID   string // Unique ID to identify message
    Message         string
    Severity        Severity
    Info            map[string]interface{} // Additional information for logging or debugging
}
```